src/tests/ftest/ior/ior_intercept_messages.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+import re
+import os
+from ior_test_base import IorTestBase
+
+
+class IorInterceptMessages(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Runs IOR with interception.
+
+       Look for messages provided the the library
+
+    :avocado: recursive
+    """
+
+    def test_ior_intercept_messages(self):
+        """Jira ID: DAOS-7647.
+
+        Test Description:
+            Purpose of this test is to run ior using dfuse with interception
+            library enabled and look for some debug messages provided
+            by the library.
+
+        Use case:
+            Run ior with dfuse flags set on debug
+            Look for interception library messages.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,small
+        :avocado: tags=daosio,dfuse,il
+        :avocado: tags=iorinterceptmessages
+        """
+        apis = self.params.get("ior_api", '/run/ior/iorflags/ssf/*')
+        d_il_report_value = self.params.get("value", "/run/tests/D_IL_REPORT/*")
+        intercept = os.path.join(self.prefix, 'lib64', 'libioil.so')
+
+        # Set the env locally for this test
+        # Avoiding any impact to the rest of IOR test cases
+        job_manager = self.get_ior_job_manager_command()
+        env = self.ior_cmd.get_default_env(str(job_manager), self.client_log)
+        env['D_LOG_MASK'] = 'INFO'
+        env['DD_MASK '] = 'all'
+        env['DD_SUBSYS'] = 'all'
+        env['LD_PRELOAD'] = intercept
+        #
+        # D_IL_REPORT VALUES
+        #     -1: All printed calls # This needs its own test case
+        #      0: Summary on exit
+        #      1: Print to stderr the first read call
+        #      2: Print to stderr the first 2 read calls
+        # If needed the test can mux the value of D_IL_REPORT
+        # an look only for a limited number of prints
+        #
+        env['D_IL_REPORT'] = d_il_report_value
+
+        # Summary
+        match_summary_results = []
+        # pylint: disable=anomalous-backslash-in-string
+        summary_pattern = "^\[libioil\] Performed [0-9]+ reads and [0-9]+ " \
```

Comment:
This would be better as a global, then you wouldn't have to cut it to match the line length.  Could you do \d rather than [0-9] as well?

Reply:
It would be of length 93 and only used within the test itself, I do not see the value of making it a global constant. Doing \d or [0-9] are equal.